en/docs/develop/api-policies/create-a-policy.md

```diff
@@ -0,0 +1,291 @@
+
+Follow the steps below to implement the policy:
+
+- Open the policy.bal file in the Ballerina project and fill in/modify the generated policy stubs(request, response, or fault) to suit your requirements. 
+
+### Request Flow
+ 
+```
+@mediation:RequestFlow
+public function validateRequestHeader(mediation:Context ctx, http:Request req, string headerName, string headerValue) returns http:Response|false|error|() {
+   string|http:HeaderNotFoundError header = req.getHeader(headerName);
+   if (header is http:HeaderNotFoundError) {
+    string message = string `Header ${headerName} is not found`;
+    log:printError(message);
+    return generateResponse(message, http:STATUS_BAD_REQUEST);
+   }
+   if (header != headerValue) {
+    string validationFailedMessage = string `Header validation failed. Expected ${headerValue} but found ${header}`;
+    log:printError(validationFailedMessage);
+    return generateResponse(validationFailedMessage, http:STATUS_BAD_REQUEST);
+    }
+    log:printInfo("Header validation successful");
+    return ();
+}
+function generateResponse(string message, int statusCode) returns http:Response {
+    http:Response response = new();
+    response.setTextPayload(message); 
+    response.statusCode = statusCode;
+    return response;
+}
+
+```
+
+### Response Flow
+
+```
+@mediation:ResponseFlow
+public function validateResponseHeader(mediation:Context ctx, http:Request req, http:Response res, string headerName, string headerValue) returns http:Response|false|error|() { 
+   string|http:HeaderNotFoundError header = res.getHeader(headerName);
+   if (header is http:HeaderNotFoundError) {
+       string message = string `Header ${headerName} is not found`;
+    log:printError(message);
+    return ();
+   }
+   if (header != headerValue) {
+    string validationFailedMessage = string `Header validation failed. Expected ${headerValue} but found ${header}`;
+    log:printError(validationFailedMessage);
+    return ();
+   }
+   return ();
+}
+
+```
+
+In this policy, you are not making any changes to the fault flow stub. Therefore, you can remove the fail flow stub from the policy.bal file.
+
+!!! note
+    The annotations **@mediation:RequestFlow** **@mediation:ResponseFlow** and **@mediation:FaultFlow** are bound with the keywords in the Ballerina.toml. Therefore, the changes you make to the policy stubs should reflect in the Ballerina.toml file. For example, if the policy is applicable only on the request and response paths, the annotation **@mediation:FaultFlow** can be removed from the policy. Then, you **MUST** remove the keyword, **choreo-apim-mediation-fault-flow** from the generated Ballerina.toml file. If failed to do so, the Ballerina compiler will show an error at compile time.
+
+!!! tip 
```

Comment:
No they are not best practices.. They are more like tips on how to develop a mediation policy.

Reply:
![Uploading Screenshot 2023-01-24 at 17.45.36.png…]()

Reply:
At the moment, these tips are under the **Response flow** sub topic and therefore appears to be applicable specifically to the Response flow.

Cannot view the screenshot because the upload was not complete.

In that case, shall combine the note and the tips and have it in a box with the title similar to "Developing the policy".

Reply:
Restructured.